src/main/java/codeu/controller/NotificationServlet.java

```diff
@@ -24,17 +32,34 @@ public void init() throws ServletException {
   void setUserStore(UserStore userStore) {
     this.userStore = userStore;
   }
+  /**
+   * Sets the EventStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setEventStore(EventStore eventStore){
+    this.eventStore = eventStore;
+  }
 
-//TODO Decide if other notification types beyond messages are needed
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-        String username = request.getParameter("username");
+        String username = (String)request.getSession().getAttribute("user");
         User user = userStore.getUser(username);
         Instant userLookedAtPage = Instant.now();
+        Instant lastSeenTime = user.getLastSeenNotifications();
+        UUID lastSeenId = null;
+        List<Event> events = eventStore.getAllEvents();
+        for(Event event : events){
```

Comment:
Are these events sorted old to new or the other way around? In other words, if there are more than one event with the same instance, do we get the index of the oldest or more recent event here?

Reply:
I am not entirely sure. When I do my own testing with the local server I can not get notifications to appear. When looking at the code for activity feed that also uses the EventStore it seems to sort old to new. This leads me to believe it would give the oldest event.

Reply:
Can we add a unit test to cover this case? It doesn't have to be part of this change, if you think it's ready, you can merge it and work on the test in a separate PR.